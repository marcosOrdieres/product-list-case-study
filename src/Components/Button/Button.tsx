import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const ButtonStyled = styled.div`
  cursor: pointer;
  color: white;
  padding-top: 10px;
  background-color: #212933;
  padding: 4px;
  border-radius: 5px;
  height: 50px;
  width: 200px;
  text-align: center;
  margin: 0 auto;
`;

const Text = styled.span`
  display: inline-block;
  margin: 5px 5px 0 0;
`;

const Button = ({ onClick, text }: any) => (
  <ButtonStyled className="button" onClick={onClick}>
    <Text>{text}</Text>
  </ButtonStyled>
);

Button.propTypes = {
  onClick: PropTypes.func,
  text: PropTypes.string
};

export default Button;