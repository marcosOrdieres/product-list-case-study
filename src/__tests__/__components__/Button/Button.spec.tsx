import React from 'react';
import { render } from '@testing-library/react';
import Button from '../../../Components/Button';


describe('Button', () => {
  it('should render and match styles', () => {
    const { asFragment } = render(
      <Button onClick={() => 'value'} text='Text' />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
