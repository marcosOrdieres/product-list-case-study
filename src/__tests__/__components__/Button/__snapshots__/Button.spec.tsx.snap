// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render and match styles 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBQB bLeCNO button"
  >
    <span
      class="sc-gsTEea fRRtOn"
    >
      Text
    </span>
  </div>
</DocumentFragment>
`;
